File: server/plugin/label/router/template.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var Template = new(template)

type template struct{}

func (r *template) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
		group := private.Group("label").Use(middleware.OperationRecord())
		group.POST("saveTemplate", apiTemplate.SaveTemplate)
		group.DELETE("deleteTemplate", apiTemplate.DeleteTemplate)
		group.POST("publishTemplate", apiTemplate.PublishTemplate)
		group.POST("unpublishTemplate", apiTemplate.UnpublishTemplate)
	}
	{
		group := private.Group("label")
		group.GET("loadTemplate", apiTemplate.LoadTemplate)
		group.GET("listTemplate", apiTemplate.ListTemplate)
		group.GET("listPublishedTemplate", apiTemplate.ListPublishedTemplate)
		group.GET("translateText", apiTemplate.TranslateText)
		group.GET("downloadBatchTemplate", apiTemplate.DownloadBatchTemplate)
	}
}

File: web/src/plugin/label/api/template.js
import service from '@/utils/request'

export const saveTemplate = (data) => {
  return service({
    url: '/label/saveTemplate',
    method: 'post',
    data
  })
}

export const loadTemplate = (params) => {
  return service({
    url: '/label/loadTemplate',
    method: 'get',
    params
  })
}

export const listTemplate = () => {
  return service({
    url: '/label/listTemplate',
    method: 'get'
  })
}

export const deleteTemplate = (params) => {
  return service({
    url: '/label/deleteTemplate',
    method: 'delete',
    params
  })
}

export const translateText = (params) => {
  return service({
    url: '/label/translateText',
    method: 'get',
    params
  })
}

export const publishTemplate = (params) => {
  return service({
    url: '/label/publishTemplate',
    method: 'post',
    params
  })
}

export const unpublishTemplate = (params) => {
  return service({
    url: '/label/unpublishTemplate',
    method: 'post',
    params
  })
}

export const listPublishedTemplate = () => {
  return service({
    url: '/label/listPublishedTemplate',
    method: 'get'
  })
}

export const downloadBatchTemplate = (params) => {
  return service({
    url: '/label/downloadBatchTemplate',
    method: 'get',
    params,
    responseType: 'blob'
  })
}
